configuration/file-configuration.ts

```diff
@@ -0,0 +1,56 @@
+import {readFile, accessSync, R_OK} from "fs";
+import {IConfiguration} from "./iconfiguration";
+
+export class FileConfiguration implements IConfiguration {
+    private userConfig: { [key: string]: any } = {};
+
+    /**
+     * Asynchronously initialize configuration values from the passed file.
+     *
+     * @param {string} configFilename - Name of the JSON file containing
+     * configuration preferences.
+     */
+    public async initialize(configFilename: string): Promise<void> {
+        try {
+            accessSync(configFilename, R_OK);
+        } catch (err) {
+            console.log("No user preference file found at", configFilename);
+            return undefined;
+        }
+        let userConfig: string = await new Promise<string>( (resolve, reject) => {
+            readFile(configFilename, "utf8", (err, result) =>
+                err ? reject(err) : resolve(result)
+            );
+        });
+        this.userConfig = JSON.parse(userConfig);
+        return undefined;
+    }
+
+    /**
+     * Get the value associated with the passed key.
+     *
+     * Attempt to throw an error if the keyed value type is not a string.
+     *
+     * @param {string} key - Name of variable to get.
+     * @return {string} Value of variable named by key.
+     */
+    public getString(key: string): string {
+        let val: string = this.userConfig[key];
+        if (typeof val !== "string") {
+            throw new Error(
+                    `Configuration service found value for ${key} that was not a string.`)
+        }
+        return val;
+    }
+
+    /**
+     * Get the value associated with the passed key.
+     *
+     * @param {string} key - Name of variable to get.
+     * @return {any} Value of variable named by key.
+     */
+    public getAny(key: string): any {
+        return this.userConfig[key];
```

Comment:
you should probably throw an exception here if value _is_ of type string, asking users to call getString instead.